src/polymer/polymer2-element-scanner.ts

```diff
@@ -462,20 +475,48 @@ class ElementVisitor implements Visitor {
       const className = astValue.getIdentifierName(expression.object);
       if (isPropertyNameIs && className) {
         return {
-          type: 'is',
-          className,
-          classNameSourceRange:
-              this._document.sourceRangeForNode(expression.object)!
+          kind: 'success',
+          value: {
+            type: 'is',
+            className,
+            classNameSourceRange:
+                this._document.sourceRangeForNode(expression.object)!
+          }
         };
       }
     }
-    this._warnings.push({
-      code: 'cant-determine-element-tagname',
-      message:
-          `Unable to evaluate this expression down to a definitive string tagname.`,
-      severity: Severity.WARNING,
-      sourceRange: this._document.sourceRangeForNode(expression)!
-    });
-    return;
+    return {
+      kind: 'failure',
+      warning: {
+        code: 'cant-determine-element-tagname',
+        message:
+            `Unable to evaluate this expression down to a definitive string tagname.`,
+        severity: Severity.WARNING,
+        sourceRange: this._document.sourceRangeForNode(expression)!
+      }
+    };
   }
 }
+
```

Comment:
Why the type definitions at the bottom of the file? I thought we always defined at the top of the file unless they were used in a single specific place.